rfpremier/ah-django
# #164857168-Add push notification capability

### What does this PR do?
 - [x] Coverts the server from WSGI To ASGI 
 - [x] Makes it possible to add push notifications through web sockets

### Task to be completed
 - [x] Add the ability to send real-time push notifications through web sockets

### How can this be manually tested?
 - Additional requirements redis server [Tutorial](https://www.digitalocean.com/community/tutorials/how-to-install-and-secure-redis-on-ubuntu-18-04)
 - Clone the repo and checkout  `ft/164857168-push-notifications-server ` branch
- Install a virtual environment and activate 
- Run `pip install -r requirements.txt`
 - Run `python manage.py makemigrations`
 - Run `python manage.py migrate`
- Run the dev server `python manage.py runserver`
- install a WebSocket client of choice or a simple javascript file to will do 
[Chrome app used to test](https://chrome.google.com/webstore/detail/simple-websocket-client/pfdhoblngboilpfeibdedpjgfnlcodoo), [Another chrome exte](https://chrome.google.com/webstore/detail/browser-websocket-client/mdmlhchldhfnfnkfmljgeinlffmdgkjo/related?hl=en)

on your selected client enter the following url and hit connect/open  `ws://127.0.0.1:8000/api/notifications/new_article_notification/`
- use postman to add a new article using `POST api/articles/`
```
{"article": {
                "title": "This is the article title",
                "description": "This is the article description",
                "body": "This is the article body",
                "image_url": "https://imageurl.com"
            }
            
            }
``` 
on your websocket client you will be able to see a json object showing the new article notification
### Any background context
N/A
 ### PT stories relevant to this PR?
 [#164857168](https://www.pivotaltracker.com/n/projects/2321565/stories/164857168)

### Screenshots
N/A
